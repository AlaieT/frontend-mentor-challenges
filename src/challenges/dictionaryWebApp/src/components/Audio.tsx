import React from "react";

import type { AudioProps } from "../types";

import { ReactComponent as PlayIcon } from "../assets/images/icon-play.svg";

import styles from "../styles/components/audio.module.scss";

const Audio = ({ src }: AudioProps) => {
  const audioRef = React.useRef<HTMLAudioElement | null>(null);

  const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
    e.preventDefault();

    if (e.code === "Space") audioRef.current?.play();
  };

  const handleOnClick = (e: React.MouseEvent<HTMLDivElement>) => {
    e.preventDefault();

    audioRef.current?.play();
  };

  return (
    <div
      id={styles.audio}
      role="button"
      tabIndex={0}
      onClick={handleOnClick}
      onKeyDown={handleKeyDown}
    >
      <PlayIcon width={75} height={75} />
      <audio ref={audioRef}>
        <source src={src} type="audio/mp3" />
      </audio>
    </div>
  );
};

export default Audio;
