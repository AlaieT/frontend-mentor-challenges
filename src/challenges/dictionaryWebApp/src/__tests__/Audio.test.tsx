import React from "react";
import { render, cleanup } from "@testing-library/react";

import Audio from "../components/Audio";

afterEach(cleanup);

describe("Audio", () => {
  describe("correct render", () => {
    it("should match snapshot", () => {
      expect(
        render(<Audio src="https://some_audio.mp3" />).asFragment()
      ).toMatchSnapshot();
    });
  });
});
