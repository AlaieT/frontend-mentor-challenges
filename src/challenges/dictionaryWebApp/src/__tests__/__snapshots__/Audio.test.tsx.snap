// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio correct render should match snapshot 1`] = `
<DocumentFragment>
  <div
    id="audio"
    role="button"
    tabindex="0"
  >
    <div
      height="75"
      width="75"
    />
    <audio>
      <source
        src="https://some_audio.mp3"
        type="audio/mp3"
      />
    </audio>
  </div>
</DocumentFragment>
`;
